playground/src/components-provider/components-provider.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018 The AMPHTML Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const COMPONENTS_URL = '/playground/amp-component-versions';
+
+export function createComponentsProvider() {
+  return new ComponentsProvider(window);
+}
+
+class ComponentsProvider {
+
+  constructor(win) {
+    this.componentsMap = {};
+    this.win = win;
+    this.win.requestIdleCallback(() => this._fetchComponents());
+  }
+
+  getComponents() {
```

Comment:
suggestion: rename to simply get as the class has only one functionality.

Reply:
Done